cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -60,14 +60,13 @@ func (s *server) Start(ctx context.Context, host component.Host) error {
 	if err != nil {
 		return fmt.Errorf("cannot find primary storage %s: %w", s.config.Storage.TracesPrimary, err)
 	}
+	f = storage.NewDecoratorFactory(f, queryMetricsFactory)
```

Comment:
@yurishkuro The change was relatively easy to do for v2 but v1 is a bit more complicated.
- The storageFactory initialized [here](https://github.com/jaegertracing/jaeger/blob/main/cmd/all-in-one/main.go#L58) is a meta factory so its not easy to cast it to a `storage.Factory` because there are additional methods that are used. 
- We could decorate the factory [here](https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/factory.go#L111) but this will require having the constructor take `metrics.Factory` (`Initialize` also takes a `metrics.Factory`)

Do you have any thoughts on how we should proceed here?

Reply:
in v2 the storage extension performs the switching by storage type. In v1 I think we need to pass metrics.Factory to 
```
storageFactory, err := storage.NewFactory(storage.FactoryConfigFromEnvAndCLI(os.Args, os.Stderr))
```

Reply:
@yurishkuro Is it fine for [this](https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/factory.go#L92) function to take `metrics.Factory`?

Reply:
yes. The Factory struct already has metricsFactory field, you can move that field into FactoryConfig

Reply:
Sounds good. We do set the `metrics.Factory` field [here](https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/factory.go#L139) however. Should we remove that and move it into the constructor?

Reply:
or just pass MF explicitly

Reply:
Do we need the meta-factory's Initialize to be the same as individual factories? I would rather pass metrics & logger to its constructor

Reply:
The meta-factory itself implements `storage.Factory` so I would guess that it does not need to be the same

Reply:
@yurishkuro Do we want to do that in this PR or should we create a follow-up for that. This PR might already get large if we’re changing the constructor.

Reply:
ok to split but then I would remove the wiring part from this PR, just introduce the decorator as unused, and then wire it in the next PR.

Reply:
@yurishkuro I had a couple of follow-ups about the approach you mentioned: 
- If we take an unqualified `metrics.Factory` - how do we persist the `query` namespace that we have at the moment? 
- Where should the initialization of the `metrics.Factory` happen? Should it happen in `NewFactory` or in `Initialize` (it is currently happening in `Initialize`)?
- Or do we want to separate the metrics factory that is held inside the meta-factory from the metrics factory that is passed down to the children factories?

Reply:
1. the decorator factory can add query namespace when creating decorator
2. move to constructor and ignore the one passed to Initialize. Ideally we should remove metrics/logger from Initialize altogether and only pass them in constructor. I am not sure why it was not done that way initially.

Reply:
@yurishkuro Sounds good. What're your thoughts on wiring everything up in this PR and then adding the annotations in the next PR since that is a change in behaviour?

Reply:
ok

Reply:
which approach? I thought we settled on not changing interfaces.

Reply:
@yurishkuro In `NewFactory`, if I do something like
```go
	for t := range uniqueTypes {
		ff, err := f.getFactoryOfType(t)
		if err != nil {
			return nil, err
		}
		f.factories[t] = storagemetrics.NewDecoratorFactory(ff, mf)
	}
```
Then this doesn't work because some factories implement interfaces other than `storage.Factory` which get lost if I have all of them use the decorator.

Reply:
does it mean we go back to the idea that each factory should just use the decorator explicitly in their CreateSpanReader functions?

Reply:
please stamp #6269

Reply:
@yurishkuro That approach didn't work in the last PR because we didn't want the `query` metric namespace to bleed into the storage factories. Could we keep the decorator factory and use it just for v2?